add Work-to-Work relationship documentation

elrayle opened the pull request:
Adds documentation for the Work-to-Work relationship.

Also refactors the layout of each relationship to avoid repeating statements and code and to make each characteristic (where defined, where tested, each direction of the relationship) easier to see at a glance.  Code is separated out into code blocks instead of being inline.

## Examples yard doc layout after refactor:

### In the work

![image](https://user-images.githubusercontent.com/6855473/153662413-76c13619-7535-443a-8ec5-66a7f984b097.png)

### In the admin set

![image](https://user-images.githubusercontent.com/6855473/153662488-a165cc6d-a4a2-4717-9f8c-af35c2bf020f.png)

cjcolvar reviewed (commented):
I think this is looking good overall!

I'm wondering if it is possible to use `*Something*` instead of `<b>Something</b>` and still retrain the same appearance in the generated yard docs.  I think the former is easier to read when reading the yard docs inline in the file.

elrayle commented:
I don't know why, but yard doesn't always like `*something*`.  It appears that:
* `*something*` will correctly become **something**
* but `*something with multiple words*` becomes \*something with multiple words\*

elrayle commented on app/models/hyrax/work.rb:
```diff
@@ -7,38 +7,61 @@ module Hyrax
+  # * <b>Work to child Work:</b> (0..m)  A work can have many child works.
+  #   * Add a child work to a work using:
+  #       Hyrax::Transactions::Container['work_resource.add_to_parent']
+  #         .call(child_work, parent_id: parent_work.id, user: current_user)
```
I'm not sure about recommending calling a transaction step outside the context of the full transaction.

### Potential Refactor Considerations

On the plus side, use of a transaction step, even outside of a full transaction, allows for the code to define what it means to add a child work to a work to be in only one place in the code.

Alternatives:  Collections have a service that adds/removes members of the collection.  Something similar could be added for works members.

Right now, the transaction to add a collection does not call the collection service.  And the collection service does not call the transaction.  It would be better if only one of those did the work and the other called to it.  I would tend to lean toward having the transaction call the service, but this is not a strong preference.

Interested in what others think.

If it's not a straight forward decision, I can break this out into an issue for further discussion.

cjcolvar replied:
I was thinking the same and thought that this is what works now, but can be updated once there is a better pattern.
+1 to a separate issue for figuring out a better pattern.

elrayle commented:
### Switched to github style markdown.

I looked at existing yard doc in other classes and modules (e.g. lib/hyrax/publish.rb) and the ones with significant yard doc seem to already be using github style markdown.

To declutter for reading in code, I removed the bold text.  I also changed code blocks to use @example.

To run locally with github markdown:

```
gem install yard
yard doc -m markdown
```

If it complains about not understanding `-m markdown`, you may need to install it.

```
gem install redcarpet
```

The resulting generated doc is a little different.  Primarily, it moves examples under all the descriptive test.  Since that seems to be what would be expected for yard doc, that seems ok.

![image](https://user-images.githubusercontent.com/6855473/153679618-2a585ff5-ad60-46e8-b06c-cb3fdc032454.png)

cjcolvar reviewed (approved):
This looks great!  Switching to github markdown to be consistent is a good catch.
